### PR TITLE
Fix for #438 incorrect manifest version

### PR DESCRIPTION
**Pull request by IainSAP**
Fix for #438 

- Finds the nearest matching manifest version if no exact match found
- Return the highest version (semantically) if the ui5 version is greater than the highesy
- Adds tests

### CODE REVIEW

File: packages/ui5-application-writer/src/data/defaults.ts [hunk 1]
```diff
@@ -132,23 +132,44 @@ function getMinUI5Version(ui5Version: string, minUI5Version?: string) {
...
-        (minUI5Version &&
-            (versionToManifestDescMapping as Record<string, string>)[
-                `${semVer.major(minUI5SemVer)}.${semVer.minor(minUI5SemVer)}`
-            ]) ??
-        UI5_DEFAULT.MANIFEST_VERSION
-    );
+function getManifestVersion(ui5Version: string, manifestVersion?: string): string {
+    const ui5SemVer = semVer.coerce(ui5Version)!;
+
+    const getClosestVersion = (ui5SemVer: semVer.SemVer) => {
```
Comment (tobiasqueck) [inline packages/ui5-application-writer/src/data/defaults.ts:146]: Minor code smell: use of the same variable name.
Doesn't cause an issue but might be confusing when reading the code.

File: packages/ui5-application-writer/src/data/defaults.ts [hunk 2]
```diff
@@ -132,23 +132,44 @@ function getMinUI5Version(ui5Version: string, minUI5Version?: string) {
...
+ * Snapshots are handled by coercion to proper versions. If the version does not exist as an exact match
+ * the nearest version lower will be used.
  *
- * @param minUI5Version - the ui5 version to be used to map to the manifest descriptor version
+ * @param ui5Version - the ui5 version to be used to map to the manifest descriptor version
  * @param manifestVersion - optional manifest descriptor version to be used if provided
  * @returns - the manifest descriptor version
  */
-function getManifestVersion(minUI5Version: string, manifestVersion?: string): string {
-    const minUI5SemVer = semVer.coerce(minUI5Version)!;
```
Comment (tobiasqueck) [inline packages/ui5-application-writer/src/data/defaults.ts:143]: Any chance you can change the import to specifically import the functions you need and separately the types using `import type`

File: packages/ui5-application-writer/src/data/defaults.ts (hunk 1 above)
Comment (tobiasqueck) [inline packages/ui5-application-writer/src/data/defaults.ts:146]: Could you please add a short jsdoc comment what this function does? I assume it finds the highest version lower than the provided version?

**Review by tobiasqueck — CHANGES_REQUESTED** [review]
Minor suggestions for improvements

File: packages/ui5-application-writer/src/data/defaults.ts (hunk 2 above)
Comment (devinea) [reply packages/ui5-application-writer/src/data/defaults.ts:143]: Odd that should be enforced by eslint rule https://github.com/SAP/open-ux-tools/blob/main/.eslintrc#L73 🤔

**Review by devinea — DISMISSED** [review]

File: packages/ui5-application-writer/src/data/defaults.ts (hunk 2 above)
Comment (IainSAP) [reply packages/ui5-application-writer/src/data/defaults.ts:143]: Done

File: packages/ui5-application-writer/src/data/defaults.ts (hunk 1 above)
Comment (IainSAP) [reply packages/ui5-application-writer/src/data/defaults.ts]: DOne

Comment (IainSAP) [reply packages/ui5-application-writer/src/data/defaults.ts]: Done

**Comment by IainSAP** [comment]
Thx @tobiasqueck @devinea - all comments addressed.

**Review by tobiasqueck — APPROVED** [review]
Thanks for addressing my comments.

**Review by devinea — APPROVED** [review]

**Pull request merged**